release notes/v0.42.0.md

```diff
@@ -0,0 +1,151 @@
+
+const CLOSED_STATE = 3
+
+export default function () {
+    var url = "ws://localhost:10000";
+    var params = { "tags": { "my_tag": "hello" } };
+
+    let ws = new WebSocket(url, null, params)
+    
+    ws.onopen = () => {
+        console.log('connected')
+        ws.send(Date.now().toString())        
+    }
+
+    let intervalId = setInterval(() => {
+        ws.ping();
+        console.log("Pinging every 1 sec (setInterval test)")
+    }, 1000);
+
+    let timeout1id = setTimeout(function () {
+        console.log('2 seconds passed, closing the socket')
+        clearInterval(intervalId)
+        ws.close()
+        
+    }, 2000);
+
+    ws.onclose = () => {
+        clearTimeout(timeout1id);
+
+        console.log('disconnected')
+    }
+
+    
+    ws.onping = () => {
+        console.log("PING!")
+    }
+
+    ws.onpong = () => {
+        console.log("PONG!")
+    }
+
+    // Multiple event handlers on the same event
+    ws.addEventListener("pong", () => {
+        console.log("OTHER PONG!")
+    })
+
+    ws.onmessage = (m) => {
+        let parsed = parseInt(m.data, 10)
+        if (Number.isNaN(parsed)) {
+            console.log('Not a number received: ', m.data)
+
+            return
+        }
+
+        console.log(`Roundtrip time: ${Date.now() - parsed} ms`);        
+
+        let timeoutId = setTimeout(function() {
+            if (ws.readyState == CLOSED_STATE) {
+                console.log("Socket closed, not sending anything");
+
+                clearTimeout(timeoutId);
+                return;
+            }
+
+            ws.send(Date.now().toString())
+        }, 500);
+    }   
+
+    ws.onerror = (e) => {
+        if (e.error != "websocket: close sent") {
+            console.log('An unexpected error occurred: ', e.error);
+        }
+    };  
+};
+```
+</details>
+
+You can find documentation and other examples of using the module in the [docs](https://k6.io/docs/javascript-api/k6-experimental/websockets).
+
+### Wildcard support for `hosts` option [#2747](https://github.com/grafana/k6/pull/2747)
+
+Thanks to the great effort from @eugercek, the [hosts](https://k6.io/docs/using-k6/k6-options/reference/#hosts) option now accepts domains that contain a wildcard.
+
+```js
+export const options = {
+  hosts: {
+    '*.k6.io': '1.2.3.4',
+  },
+}
+```
+
+### Enhancements, and UX improvements
+
+- [#2660](https://github.com/grafana/k6/pull/2660) Pre-loading the operating system TLS certificates. Thanks, @tbourrely.
+- [#2773](https://github.com/grafana/k6/pull/2773) Ensures that JavaScript runtime makes the only copy of the exports for each module.
+- [#2791](https://github.com/grafana/k6/pull/2791) Runs `setup()` and `teardown()` only if they are defined in a script.
+
+## Breaking changes
+
+- [#2712](https://github.com/grafana/k6/pull/2712) `k6/ws` return HTTP response for failed connections instead of an undefined behavior. Thanks, @brietaylor.
```

Comment:
```suggestion
- [#2712](https://github.com/grafana/k6/pull/2712) `k6/ws` returns an HTTP response for failed connections instead of an undefined behavior. Thanks, @brietaylor.
```